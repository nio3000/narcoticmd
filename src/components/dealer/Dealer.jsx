import React, { Component } from 'react'
import Cocaine from "../cocaine/Cocaine";


export class Dealer extends Component {
  constructor( props ) {
    super(props);

    this.state = {
      text: ''
    };

    this.onTextChange = this.onTextChange.bind(this);
    this.onSelectText = this.onSelectText.bind(this);
  }

  onSelectText( event ) {
    let selectionStart = event.target.selectionStart;
    let selectionEnd = event.target.selectionEnd;

    if(selectionStart !== selectionEnd) {
      console.log(`${selectionStart} >><< ${selectionEnd}`);
    }
  }

  onTextChange( event ) {
    this.setState({text: event.target.value})
  }

  render() {
    return (
      <div className="Dealer" style={style.container}>
        <Cocaine visible={true} />
        <textarea
          value={this.state.text}
          onChange={this.onTextChange}
          onSelect={this.onSelectText}
          className="textarea"
          style={style.textarea}
        />
      </div>
    );
  }
};

export const style = {
  container: {
    height: '70%',
    margin: 'auto',
    width: '80%'
  },
  textarea: {
    border: '2px solid #999',
    height: '100%',
    width: '100%',
  }
};


export default Dealer;