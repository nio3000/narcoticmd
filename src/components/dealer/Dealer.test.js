import React from 'react';
import TestUtils from 'react-dom/test-utils'
import Dealer from './Dealer';

import renderer from 'react-test-renderer';

describe('Dealer', () => {

  test('renders correctly', () => {
    const component = renderer.create(
      <Dealer />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  //TODO replace snaphosts with Enzyme unit testing
  test('onChange', () =>{
    const dealer = TestUtils.renderIntoDocument(<Dealer/>);
    const textarea = TestUtils.findRenderedDOMComponentWithTag(dealer, 'textarea');
    textarea.value = 'elo';
    TestUtils.Simulate.change(textarea);

    expect(textarea).toMatchSnapshot();
  })
});
