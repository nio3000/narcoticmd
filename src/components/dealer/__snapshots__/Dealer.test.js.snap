// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dealer onChange 1`] = `
<textarea
  class="textarea"
  style="border: 2px solid #999; height: 100%; width: 100%;"
>
  elo
</textarea>
`;

exports[`Dealer renders correctly 1`] = `
<div
  className="Dealer"
  style={
    Object {
      "height": "70%",
      "margin": "auto",
      "width": "80%",
    }
  }
>
  <textarea
    className="textarea"
    onChange={[Function]}
    onSelect={[Function]}
    style={
      Object {
        "border": "2px solid #999",
        "height": "100%",
        "width": "100%",
      }
    }
    value=""
  />
</div>
`;
